cmd/lotus-worker/main.go

```diff
@@ -623,6 +645,15 @@ var runCmd = &cli.Command{
 			}
 		}()
 
+		go func() {
+			<-workerApi.Done()
+			log.Warn("Shutting down...")
```

Comment:
We should let lotus-miner know that the worker is shutting down here before stopping worker API server - this can be one in one of two ways:
* Wait for 20-30 seconds here - workers are pinged every 10s (`paths.HeartbeatInterval`), and if a ping tells the lotus-miner that the worker is shutting down, it will get fully unregistered instead of being marked as disabled (see https://github.com/filecoin-project/lotus/blob/master/storage/sealer/sched_worker.go#L120-L121)
* We could add a new API method to `lotus-miner` which would tell a `schedWorker.handleWorker` goroutine to check the worker without waiting - this is more complicated, but wouldn't require waiting as much
  * Or even add a method to explicitly disconnect / drop the worker

For now just waiting is ok, we can improve this later.

Reply:
Added the wait